HLTrigger/HLTfilters/plugins/HLTTripletMass.cc

```diff
@@ -0,0 +1,224 @@
+        filterproduct.addCollectionTag(originTag3_[i]);
+      }
+      tagOld = edm::InputTag();
+      for (trigger::size_type i3 = 0; i3 != n3; ++i3) {
+        const edm::ProductID pid(coll3[i3].id());
+        const auto& prov = iEvent.getStableProvenance(pid);
+        const std::string& label(prov.moduleLabel());
+        const std::string& instance(prov.productInstanceName());
+        const std::string& process(prov.processName());
+        edm::InputTag tagNew(edm::InputTag(label, instance, process));
+        if (tagOld.encode() != tagNew.encode()) {
+          filterproduct.addCollectionTag(tagNew);
+          tagOld = tagNew;
+        }
+      }
+    }
+
+    return true;
+  } else
+    return false;
+}
+
+// ------------ method called to produce the data  ------------
+template <typename T1, typename T2, typename T3>
+bool HLTTripletMass<T1, T2, T3>::hltFilter(edm::Event& iEvent,
+                                           const edm::EventSetup& iSetup,
+                                           trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  // All HLT filters must create and fill an HLT filter object,
+  // recording any reconstructed physics objects satisfying (or not)
+  // this HLT filter, and place it in the Event.
+  bool accept(false);
+
+  std::vector<T1Ref> coll1;
+  std::vector<T2Ref> coll2;
+  std::vector<T3Ref> coll3;
+
+  int n(0);
+  if (getCollections(iEvent, coll1, coll2, coll3, filterproduct)) {
+    T1Ref r1;
+    T2Ref r2;
+    T3Ref r3;
+
+    reco::Particle::LorentzVector dauC_p4, dauA_p4, dauB_p4, dauAB_p4, mother_p4;
+    for (unsigned int i1 = 0; i1 != coll1.size(); i1++) {
+      r1 = coll1[i1];
+      dauA_p4 = reco::Particle::LorentzVector(r1->px(), r1->py(), r1->pz(), r1->energy());
+      unsigned int i2 = is1and2Same ? i1 + 1 : 0;
+      for (; i2 != coll2.size(); i2++) {
+        r2 = coll2[i2];
+        dauB_p4 = reco::Particle::LorentzVector(r2->px(), r2->py(), r2->pz(), r2->energy());
+        dauAB_p4 = dauA_p4 + dauB_p4;
+
+        unsigned int i3 = is2and3Same ? i2 + 1 : 0;
+        for (; i3 != coll3.size(); i3++) {
+          r3 = coll3[i3];
+          dauC_p4 = reco::Particle::LorentzVector(r3->px(), r3->py(), r3->pz(), r3->energy());
+          if (reco::deltaR(dauAB_p4, dauC_p4) > max_DR_) {
+            continue;
+          }
+          mother_p4 = dauC_p4 + dauAB_p4;
+          for (unsigned int j = 0; j < max_InvMass_.size(); j++) {
+            if ((mother_p4.mass() >= min_InvMass_[j]) and (mother_p4.mass() < max_InvMass_[j])) {
+              accept = true;
+            }
+
+            if (accept)
+              break;
+          }
+          if (accept) {
+            n++;
+            filterproduct.addObject(triggerType1_, r1);
+            filterproduct.addObject(triggerType2_, r2);
+            filterproduct.addObject(triggerType3_, r3);
+            break;
+          }
+        }
+        if (accept)
+          break;
```

Comment:
Why do `break` in L209 and L213? Doesn't one want to find and count all the tri-candidates that pass the selection?

Reply:
yes , removing `break` makes sense since `min_N_` is also a parameter to the filter . I will update it

Reply:
done !